pyexchange/leverjfutures.py

```diff
@@ -0,0 +1,412 @@
+        assert(isinstance(pair, str))
+        assert(isinstance(is_sell, bool))
+        assert(isinstance(price, Wad))
+        assert(isinstance(amount, Wad))
+
+        orderInstrument = self.get_product(pair)
+        side = "sell" if is_sell else "buy"
+        price = str(price)
+        triggerPrice = str(triggerPrice)
+        quantity = str(amount)
+        order = self.createNewOrder(side, price, triggerPrice, quantity, orderInstrument, orderType)
+        self.logger.info(f'LEVERJ: order is {order}')
+        return self._http_authenticated("POST", "/futures/api/v1", "/order", [order])[0]['uuid']
+
+    def cancel_order(self, order_id: str) -> bool:
+        assert(isinstance(order_id, str))
+
+        result = self._http_authenticated("DELETE", "/futures/api/v1", f"/order/{order_id}", None)
+
+        if order_id != result[0][0]:
+            return False
+
+        return True
+
+    def cancel_all_orders(self) -> List:
+        result = []
+
+        orders = self.get_all_orders()
+
+        for order in orders:
+            order_id = order['uuid']
+            result.append(self.cancel_order(order_id))
+
+        return result
+
+    def _http_authenticated(self, method: str, api_path: str, resource: str, body):
+        assert(isinstance(method, str))
+        assert(isinstance(api_path, str))
+        assert(isinstance(resource, str))
+        assert(isinstance(body, dict) or (body is None) or (body, list))
+
+        data = json.dumps(body, separators=(',', ':'))
+        nonce = int(time.time()*1000)
+
+        params = {
+            'method': method,
+            'uri': resource,
+            'nonce': nonce
+            }
+
+
+        if body is not None:
+            params['body'] = body
+
+        payload = str(nonce)
+        signature = self._create_signature(payload)
+
+        v, r, s = to_vrs(signature)
+
+        auth_header = f"NONCE {self.account_id}.{self.api_key}"\
+            f".{v}"\
+            f".{bytes_to_hexstring(r)}"\
+            f".{bytes_to_hexstring(s)}"
+
+        headers={ "Authorization": auth_header, "Nonce": str(nonce) }
+        if body is not None:
+            headers["Content-Type"] = "application/json"
+
+        return self._result(requests.request(method=method,
+                                             url=f"{self.api_server}{api_path}{resource}",
+                                             data=data,
+                                             headers=headers,
+                                             timeout=self.timeout))
+
+    def _create_signature(self, params: str) -> str:
+        assert(isinstance(params, str))
+
+        return eth_sign(bytes(params, 'utf-8'), self.web3, self.api_secret, False, Address(self.account_id))
+
+    def _result(self, result) -> Optional[dict]:
+        if not result.ok:
+            raise Exception(f"Leverj API invalid HTTP response: {http_response_summary(result)}")
```

Comment:
Raising base exception class is usually a bad practice; `RuntimeError` might be better here.